internal/flagtypes/dryrun.go

```diff
@@ -0,0 +1,71 @@
+package flagtypes
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// ensure they conform to the interfaces
+var dryRun = DryRunNone
+var _ pflag.Value = &dryRun
+
+// DryRun is an enum flag for setting dry-run.
+type DryRun string
+
+const (
+	// DryRunNone disables dry-run. The build will be performed as usual
+	DryRunNone DryRun = "none"
+	// DryRunClient only logs what would be run, without contacting Kubernetes
+	DryRunClient DryRun = "client"
+	// DryRunServer submits server-side dry-run requests to Kubernetes
+	DryRunServer DryRun = "server"
```

Comment:
nit: end sentence with `.`

```suggestion
	// DryRunNone disables dry-run. The build will be performed as usual.
	DryRunNone DryRun = "none"
	// DryRunClient only logs what would be run, without contacting Kubernetes.
	DryRunClient DryRun = "client"
	// DryRunServer submits server-side dry-run requests to Kubernetes.
	DryRunServer DryRun = "server"
```